driver-reactive-streams/src/test/functional/com/mongodb/reactivestreams/client/syncadapter/SyncGridFSBucket.java

```diff
@@ -0,0 +1,376 @@
+                                 final GridFSDownloadOptions options) {
+        toOutputStream(wrapped.downloadToPublisher(unwrap(clientSession), filename, options), destination);
+    }
+
+    @Override
+    public GridFSFindIterable find() {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public GridFSFindIterable find(final Bson filter) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public GridFSFindIterable find(final ClientSession clientSession) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public GridFSFindIterable find(final ClientSession clientSession, final Bson filter) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void delete(final ObjectId id) {
+        delete(new BsonObjectId(id));
+    }
+
+    @Override
+    public void delete(final BsonValue id) {
+        Mono.from(wrapped.delete(id)).block(TIMEOUT_DURATION);
+    }
+
+    @Override
+    public void delete(final ClientSession clientSession, final ObjectId id) {
+        delete(clientSession, new BsonObjectId(id));
+    }
+
+    @Override
+    public void delete(final ClientSession clientSession, final BsonValue id) {
+        Mono.from(wrapped.delete(unwrap(clientSession), id)).block(TIMEOUT_DURATION);
+    }
+
+    @Override
+    public void rename(final ObjectId id, final String newFilename) {
+        rename(new BsonObjectId(id), newFilename);
+    }
+
+    @Override
+    public void rename(final BsonValue id, final String newFilename) {
+        Mono.from(wrapped.rename(id, newFilename)).block(TIMEOUT_DURATION);
+    }
+
+    @Override
+    public void rename(final ClientSession clientSession, final ObjectId id, final String newFilename) {
+        rename(clientSession, new BsonObjectId(id), newFilename);
+    }
+
+    @Override
+    public void rename(final ClientSession clientSession, final BsonValue id, final String newFilename) {
+        Mono.from(wrapped.rename(unwrap(clientSession), id, newFilename)).block(TIMEOUT_DURATION);
+    }
+
+    @Override
+    public void drop() {
+        Mono.from(wrapped.drop()).block(TIMEOUT_DURATION);
+    }
+
+    @Override
+    public void drop(final ClientSession clientSession) {
+        Mono.from(wrapped.drop(unwrap(clientSession))).block(TIMEOUT_DURATION);
+    }
+
+    private void toOutputStream(final GridFSDownloadPublisher downloadPublisher, final OutputStream destination) {
+        Flux.from(downloadPublisher).toStream().forEach(byteBuffer -> {
+            try {
+                byte[] bytes = new byte[byteBuffer.remaining()];
+                byteBuffer.get(bytes);
+                destination.write(bytes);
+            } catch (IOException e) {
+                throw new MongoGridFSException("IOException when reading from the OutputStream", e);
+            }
+        });
```

Comment:
[A bug in the implementation of our `downloadToStream` methods?]

The `toOutputStream` method is used to implement `com.mongodb.client.gridfs.GridFSBucket.downloadToStream` methods. Those methods do not specify whether the destination `OutputStream` is flushed or not, but the [example in our documentation](https://docs.mongodb.com/drivers/java/sync/current/fundamentals/gridfs/#download-a-file-to-an-output-stream) clearly does not flush the stream, implying that users do not need to flush it, i.e., it is flushed by `downloadToStream` methods. I looked at `com.mongodb.client.gridfs.GridFSBucketImpl.downloadToStream` and discovered that our real implementation does not flush the stream just like the `SyncGridFSBucket` does not. This is disturbing, because users expect streams to be flushed based on the aforementioned example, which means that depending on the `OutputStream` implementation, a part of the read data may be lost.

@jyemin I would like to know your opinion before filing a bug about our `downloadToStream` methods.

Reply:
I'm not sure it should be the driver's responsibility to flush the stream.  Regarding the documentation, it's true it doesn't flush the stream, but it does close it.  There seems to be a lot of conflicting opinions about whether closing a file outputstream stream is sufficient to also flush it (it would seem weird if it didn't, but it's not explicitly stated in the javadoc).

Anyway, let's discuss in a different channel than this PR.